apps/src/studio/blocks.js

```diff
@@ -1845,6 +1846,29 @@ exports.install = function (blockly, blockInstallOptions) {
   };
 
 
+  blockly.Blocks.studio_whenSpriteAndGroupCollideSimple = {
+    // Block to handle event when a sprite collides with any sprite in a group
+    helpUrl: '',
+    init: function () {
+      this.setHSV(140, 1.00, 0.74);
+
+      var dropdown1 = spriteNumberTextDropdown(msg.whenSpriteN);
+      var dropdown2 = createSpriteGroupDropdown(msg.collidesWithAnySpriteName);
+      this.appendDummyInput()
+          .appendTitle(dropdown1, 'SPRITE')
+          .appendTitle(dropdown2, 'SPRITENAME');
+
+      this.setPreviousStatement(false);
+      this.setNextStatement(true);
+      this.setTooltip(msg.whenSpriteAndGroupCollideTooltip());
+    },
+  };
+
+  generator.studio_whenSpriteAndGroupCollideSimple = function () {
+    return '';
```

Comment:
Yep, event blocks don't need any code, studio.js finds the attached blocks and runs them when necessary.

Though it looks like most of the other event blocks use `generator.studio_eventHandlerPrologue`, which just generates a newline. Updated to use that.